app/lib/helpers/reportStatusHelpers.tsx

```diff
@@ -0,0 +1,36 @@
+import { DateTime } from "luxon";
+
+export const daysUntilReportDue = (reportDueDate) => {
+  const diff = DateTime.fromISO(reportDueDate, {
+    setZone: true,
+    locale: "en-CA",
+  }).diff(
+    // Current date without time information
+    DateTime.now().setZone("America/Vancouver").startOf("day"),
+    "days"
+  );
+
+  return Math.ceil(diff.days);
+};
+
+export const isOverdue = (reportDueDate) => {
+  return reportDueDate ? daysUntilReportDue(reportDueDate) < 0 : false;
+};
+
+export const determineVariant = (reportFormChanges, isReportOverdue) => {
```

Comment:
not sure about the function name. I seems to return a status.

Reply:
The arguments should be typed, especially the first one as the type could be changed by modifying a relay query, breaking this function

Reply:
> not sure about the function name. I seems to return a status.

The `StatusBadge` component calls the prop variant. (`interface Props {
  label?: string;
  variant: "complete" | "late" | "onTrack" | "inReview" | "customText" | "none";
}`)